aws/aws-health-tools
# Automated root volume (/dev/xvda || /dev/sda1 ) Recovery for AWS_EBS_VOLUME_LOST event

This templates deploys Automated root volume recovery on AWS_EBS_VOLUME_LOST event.
Using AWS StepFunctions.

The current StateMachine flow will go like this:
1.       FindDetails about Volume
a.       Check if the volume is attached as a device or not, if not then FAIL the step.
b.       Check if the volume is attached as other than /dev/xvda or /dev/sda1, if yes then FAIL the step.
c.       Else Move to FindLatestSnapshot
At each of the FAIL step we can set it to trigger an SNS notification to notify the user to take action manually.
 
2.       FindLatestSnapshot
a.       Check if snapshot exist, if not then FAIL
b.       If Snapshot exist find the lastest one.
c.       Else Move on to StopInstance.
 
3.       Stop Instance
4.       Wait for Instance to Stop
5.       Detach RootVolume
6.       Wait Detach Volume
7.       Restore Snapshot
8.       Wait Volume Created
9.       Attach new Volume as /dev/xvda or /dev/sda1
10.   Wait Volume Atached
11.   Start Instance
Notify SNS upon success.


ReadMe coming soon
